### DISCUSSION TITLE
[Supply] Add option to name changelog to version_name instead of version_code

### DISCUSSION THREAD

**Discussion by kevinboosten** [fastlane/fastlane#19221] [Ideas]
Actually a "repost" of issue #15731 that got closed. But I'm curious about this as well. Here is the initial request:

**Motivation Behind Feature**
I have dynamic version numbers for each git commit, and only know the actual versioncode after the build has been finished. I tag the commit to release with the proper version name, and then start CI build/deploy.

Currently I can't deploy to the store, since I need to edit the changelog names with the new version_code, and do a new run, which doesn't make sense. I want to have my changelogs and code in a single commit that can be tagged and be used as reference to that release.

I don't really understand why supply uses versioncode, since changelogs are for actual user facing versions? Anyhow, it would be great to have the option to choose version name instead of version code for the filenames.

**Feature Description**
Add option to use version_name instead of version_code for changelog txt file name.

**Alternatives or Workarounds**
Maybe I can override something in fastlane? I don't know.

docs: https://docs.fastlane.tools/actions/supply/#changelogs-whats-new

**Comment by SapuSeven** [comment]
I was facing the same problem. It looks like there is no solution available yet, so I'm using the following workaround for now:

```
for f in fastlane/metadata/android/*/changelogs; do [[ -e "$f/$VERSION.txt" ]] && cp "$f/$VERSION.txt" "$f/default.txt"; done;
```

This command gets executed before fastlane in my build workflow (make sure to set `VERSION` accordingly) and just moves the appropriate changelog file to be the default.

**Comment by h-sigma** [comment]
Bing, bong. This would be great to have.
